File: backend/app/schemas/analytics.py
import uuid
from datetime import datetime

from pydantic import BaseModel


class ProgressEntry(BaseModel):
    session_id: uuid.UUID
    scenario_title: str
    completed_at: datetime
    overall_score: float


class MyProgressOut(BaseModel):
    entries: list[ProgressEntry]
    average_score: float | None


class RepStat(BaseModel):
    rep_id: uuid.UUID
    rep_name: str
    sessions_completed: int
    average_score: float


class ScenarioStat(BaseModel):
    scenario_id: uuid.UUID
    title: str
    attempts: int
    average_score: float


class CriterionStat(BaseModel):
    name: str
    average_score: float
    sample_count: int


class TeamAnalyticsOut(BaseModel):
    total_evaluated_sessions: int
    average_score: float | None
    by_rep: list[RepStat]
    by_scenario: list[ScenarioStat]
    by_criterion: list[CriterionStat]

File: frontend/src/lib/analytics.ts
import { api } from "./api";

export interface ProgressEntry {
  session_id: string;
  scenario_title: string;
  completed_at: string;
  overall_score: number;
}

export interface MyProgress {
  entries: ProgressEntry[];
  average_score: number | null;
}

export interface RepStat {
  rep_id: string;
  rep_name: string;
  sessions_completed: number;
  average_score: number;
}

export interface ScenarioStat {
  scenario_id: string;
  title: string;
  attempts: number;
  average_score: number;
}

export interface CriterionStat {
  name: string;
  average_score: number;
  sample_count: number;
}

export interface TeamAnalytics {
  total_evaluated_sessions: number;
  average_score: number | null;
  by_rep: RepStat[];
  by_scenario: ScenarioStat[];
  by_criterion: CriterionStat[];
}

export const analyticsApi = {
  myProgress: () => api.get<MyProgress>("/analytics/my-progress"),
  team: () => api.get<TeamAnalytics>("/analytics/team"),
};
